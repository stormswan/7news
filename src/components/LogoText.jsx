import styled from 'styled-components'

const LogoText = styled.div.attrs ({
  size: props => props.size || '3em',
  color: props => props.color || 'white',
})`
    color: ${props => props.color};
    font-size: ${props => props.size};

    @media (max-width: 576px) {
        font-size: 1.8em;
    };

    @media (min-width: 992px) {
        font-size: 4em;
    };

    @media (min-width: 1200px) {
        font-size: 5em;
    };
`

export default LogoText