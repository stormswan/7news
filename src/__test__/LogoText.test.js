import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import LogoText from '../components/LogoText'


test('Shapshot', () => {
  const logoText = renderer.create( <LogoText>Home and Away</LogoText>).toJSON()
  expect(logoText).toMatchSnapshot()
})

test('should change font size at differnet screen size', () => {
  const logoText = renderer.create( <LogoText>Home and Away</LogoText>).toJSON()
  expect(logoText).toHaveStyleRule('font-size', '4em', {
    media: '(min-width: 992px)'
  })

  expect(logoText).toHaveStyleRule('font-size', '5em', {
    media: '(min-width: 1200px)'
  })

  expect(logoText).toHaveStyleRule('font-size', '1.8em', {
    media: '(max-width: 576px)'
  })
});


test('should work with default value', () => {
  const logoText = renderer.create( <LogoText>Home and Away</LogoText>).toJSON()
  expect(logoText).toHaveStyleRule('color','white')
  expect(logoText).toHaveStyleRule('font-size','3em')
  expect(logoText).toHaveStyleRule('margin-left','0.85em')
});

test('should work with props value', () => {
  const color = 'green'
  const size ='2em'
  const logoText = renderer.create( <LogoText color={color} size={size}>Home and Away</LogoText>).toJSON()
  expect(logoText).toHaveStyleRule('color',color)
  expect(logoText).toHaveStyleRule('font-size',size)
});