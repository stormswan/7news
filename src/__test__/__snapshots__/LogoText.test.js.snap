// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shapshot 1`] = `
.c0 {
  margin-left: 0.85em;
  color: white;
  font-size: 3em;
}

@media (max-width:576px) {
  .c0 {
    font-size: 1.8em;
  }
}

@media (min-width:992px) {
  .c0 {
    font-size: 4em;
  }
}

@media (min-width:1200px) {
  .c0 {
    font-size: 5em;
  }
}

<div
  className="c0"
  color="white"
  size="3em"
>
  Home and Away
</div>
`;
